ui/apps/platform/src/hooks/useURLSort.test.js

```diff
@@ -176,14 +176,190 @@ describe('useURLSort', () => {
             expect(result.current.sortOption.field).toEqual('Bogus');
             expect(result.current.sortOption.reversed).toEqual(false);
             expect(result.current.getSortParams('Name').columnIndex).toEqual(0);
-            expect(result.current.getSortParams('Name').sortBy.index).toEqual(undefined);
+            expect(result.current.getSortParams('Name').sortBy.index).toEqual(-1);
+            expect(result.current.getSortParams('Name').sortBy.direction).toEqual('asc');
+            expect(result.current.getSortParams('Status').columnIndex).toEqual(1);
+            expect(result.current.getSortParams('Status').sortBy.index).toEqual(-1);
+            expect(result.current.getSortParams('Status').sortBy.direction).toEqual('asc');
+            expect(result.current.getSortParams('Bogus').columnIndex).toEqual(-1);
+            expect(result.current.getSortParams('Bogus').sortBy.index).toEqual(-1);
+            expect(result.current.getSortParams('Bogus').sortBy.direction).toEqual('asc');
+        });
+    });
+
+    describe('when using URL sort with multiple sort options', () => {
+        const params = {
+            sortFields: [
+                'Name',
+                'Status',
+                ['Critical severity count', 'Low severity count'],
+                'Created date',
+            ],
+            defaultSortOption: {
+                field: 'Name',
+                direction: 'desc',
+            },
+        };
+
+        it('should get the sort options from URL by default', () => {
+            const { result } = renderHook(() => useURLSort(params), { wrapper });
+
+            expect(result.current.sortOption.field).toEqual('Name');
+            expect(result.current.sortOption.reversed).toEqual(true);
+        });
+
+        it('should change sorting fields and directions', () => {
+            const { result } = renderHook(() => useURLSort(params), { wrapper });
+
+            expect(result.current.sortOption.field).toEqual('Name');
+            expect(result.current.sortOption.reversed).toEqual(true);
+
+            // Change to single field sort
+            actAndRunTicks(() => {
+                result.current.getSortParams('Status').onSort(null, null, 'desc');
+            });
+
+            expect(result.current.sortOption.field).toEqual('Status');
+            expect(result.current.sortOption.reversed).toEqual(true);
+
+            // Change to subset of multi field sort
+            actAndRunTicks(() => {
+                result.current
+                    .getSortParams('Severity', [{ field: 'Critical severity count' }])
```

Comment:
Title Case for backend search and sort fields?
https://github.com/stackrox/stackrox/blob/master/pkg/search/options.go#L332

Reply:
Good point, even though this is a test no reason to not be consistent with the application code style.

(Although, since the BE is case-insensitive for sort fields, I was tempted to build in case insensitive checking into the UI. Better to keep it clean and consistent though.)